server/db/index.js

```diff
@@ -0,0 +1,129 @@
+const Redis = require("ioredis");
+
+const { createGame, deleteGame, getPlayers, addPlayer, getPlayerIndex, playerExists } = require('./game');
+const { getPlayerSocket, setPlayerSocket } = require('./player');
+const {
+    evaluateTrick,
+    getCurrentRound,
+    getCurrentTrick,
+    getPlayerBets,
+    getPlayerCards,
+    getTrickLeader,
+    getTrickWinners,
+    playCard,
+    setCurrentRound,
+    setCurrentTrick,
+    setPlayerBet,
+    startRound,
+    whosTurnIsIt,
+} = require('./round');
+
+const TOTAL_CARDS = 60;
+const MIN_PLAYERS = 3;
+
+
+
+const connect = async () => {
+    const redis = new Redis();
+    try {
+        await redis.connect();
+    } catch (err) { }
+    return redis;
+};
+
+
+const getGameState = async (redis, gameId) => {
+    // {}[round] -> {}[playerId] -> {bet: number, taken: number}
+    const [currentRound, currentTrick, players] = await Promise.all([
+        getCurrentRound(redis, gameId),
+        getCurrentTrick(redis, gameId),
+        getPlayers(redis, gameId),
+    ]);
+    const rounds = [];
+    for (let round = 0; round < currentRound + 1; round++) {
+        rounds.push(round);
+    }
+    const allBets = await Promise.all(rounds.map(async round => getPlayerBets(redis, gameId, round)));
+    const trickWinners = await Promise.all(rounds.map(async round => getTrickWinners(redis, gameId, round)));
+
+    const scores = {};
+    allBets.forEach((bets, round) => {
+        scores[round] = {};
+        players.forEach((playerId, playerIndex) => {
+            scores[round][playerId] = { bet: bets[playerIndex], taken: 0 };
+        });
+
+        for (const trickWinner of trickWinners[round]) {
+            scores[round][trickWinner].taken += 1;
+        }
+    });
+
+    const activePlayer = await whosTurnIsIt(redis, gameId);
+
+    return {
```

Comment:
this should also return the players cards, the trickLeader, and the cards played so far for the trick

Reply:
would adding something like below work?

```
{
  trick: {
    number: 1, // previous trick
    cards: [...],
    leader: 'name'
}
```

Reply:
I would rather have the cards as `{ username: card }` as that's how I'm storing them in the client.

I'm not sure what `// previous trick` is about?

Reply:
sorry, that should have been "previously trick" was suggesting moving the trick number from ".trick" to ".trick.number"

Reply:
> I would rather have the cards as `{ username: card }` as that's how I'm storing them in the client.

you'd don't want the card order?

Reply:
if it makes no difference on your end, can they just be 'trickNumber', 'trickCards', and 'trickLeader'? otherwise I just have to transform them all to that

Reply:
> > I would rather have the cards as `{ username: card }` as that's how I'm storing them in the client.
> 
> you'd don't want the card order?

no thats much more difficult to work with